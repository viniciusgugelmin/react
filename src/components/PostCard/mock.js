export const PostCardPropsMock = {
  title: 'Lorem ipsum title',
  body: 'Lorem ipsum body',
  id: 1,
  cover: 'img/lorem-ipsum-img.png',
};
