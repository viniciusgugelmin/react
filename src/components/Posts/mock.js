export const PostsMock = [
  {
    id: 1,
    title: 'Lorem ipsum title 1',
    body: 'Lorem ipsum body 1',
    cover: 'img/img1.png',
  },
  {
    id: 2,
    title: 'Lorem ipsum title 2',
    body: 'Lorem ipsum body 2',
    cover: 'img/img2.png',
  },
  {
    id: 3,
    title: 'Lorem ipsum title 3',
    body: 'Lorem ipsum body 3',
    cover: 'img/img3.png',
  },
];
