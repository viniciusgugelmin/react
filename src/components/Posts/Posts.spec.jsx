import { render, screen } from '@testing-library/react';
import { Posts } from './index';
import { PostsMock as posts } from './mock';

describe('<Posts />', () => {
  it('should render posts', () => {
    render(<Posts posts={posts} />);

    expect(screen.getAllByRole('heading', { name: /^Lorem ipsum title/i })).toHaveLength(3);
    expect(screen.getAllByRole('img', { name: /^Lorem ipsum title/i })).toHaveLength(3);
    expect(screen.getAllByText(/^Lorem ipsum body/i)).toHaveLength(3);
    expect(screen.getByRole('img', { name: posts[0].title })).toHaveAttribute('src', posts[0].cover);
  });

  it('should not render posts', () => {
    render(<Posts />);

    expect(screen.getByText(/^No posts found$/i)).toBeInTheDocument();
  });

  it('should match snapchot', () => {
    const { container } = render(<Posts posts={posts} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
