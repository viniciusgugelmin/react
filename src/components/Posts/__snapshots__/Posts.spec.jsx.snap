// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapchot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="Lorem ipsum title 1"
      src="img/img1.png"
    />
    <div
      class="post-content"
    >
      <h2>
        Lorem ipsum title 1
      </h2>
      <p>
        Lorem ipsum body 1
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Lorem ipsum title 2"
      src="img/img2.png"
    />
    <div
      class="post-content"
    >
      <h2>
        Lorem ipsum title 2
      </h2>
      <p>
        Lorem ipsum body 2
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Lorem ipsum title 3"
      src="img/img3.png"
    />
    <div
      class="post-content"
    >
      <h2>
        Lorem ipsum title 3
      </h2>
      <p>
        Lorem ipsum body 3
      </p>
    </div>
  </div>
</div>
`;
